qiskit/synthesis/discrete_basis/generate_basis_approximations.py

```diff
@@ -76,7 +76,7 @@ def _check_candidate_greedy(candidate, existing_sequences, tol=1e-10):
         return False
 
     for existing in existing_sequences:
-        if matrix_equal(existing.product_su2, candidate.product_su2, ignore_phase=True, atol=tol):
+        if matrix_equal(existing.product, candidate.product, ignore_phase=False, atol=tol):
```

Comment:
I'm wondering if `ignore_phase` even matters here, since the SO(3) product does not have a phase. In this case I'd rather not introduce a change

Reply:
Agreed, changed it back.